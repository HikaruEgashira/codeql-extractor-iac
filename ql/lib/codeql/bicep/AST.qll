import codeql.bicep.ast.AstNodes
import codeql.bicep.ast.Expr
import codeql.bicep.ast.Literal
