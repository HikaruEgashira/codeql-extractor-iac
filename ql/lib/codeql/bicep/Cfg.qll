private import codeql.bicep.controlflow.internal.Cfg as CfgInternal
import CfgInternal::Completion
import CfgInternal::CfgScope
import CfgInternal::CfgImpl
