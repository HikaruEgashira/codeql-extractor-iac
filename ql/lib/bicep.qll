import codeql.Locations
import codeql.files.FileSystem
import codeql.bicep.AST
