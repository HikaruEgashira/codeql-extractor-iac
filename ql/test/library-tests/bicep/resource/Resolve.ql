import bicep

query predicate resolveIdentifier(Network::VirtualNetworks vn, Network::VirtualNetworkSubnets vns) {
  vns.getParent() = vn
}

query predicate resolveResource(Compute::VirtualMachines vm, Network::NetworkInterfaces ni) {
  ni = vm.getNetworkInterfaces()
}
