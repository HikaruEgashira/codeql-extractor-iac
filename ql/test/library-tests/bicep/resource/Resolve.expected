resolveIdentifier
| sample.bicep:1:1:3:1 | VirtualNetworks Resource | sample.bicep:5:1:8:1 | ResourceDeclaration |
resolveResource
| sample.bicep:15:1:28:1 | VirtualMachines Resource | sample.bicep:10:1:13:1 | NetworkInterfaces Resource |
